ldm/invoke/generator/txt2img2img.py

```diff
@@ -38,10 +41,32 @@ def get_make_image(self, prompt:str, sampler, steps:int, cfg_scale:float, ddim_e
                 uc, c, cfg_scale, extra_conditioning_info,
                 threshold = ThresholdSettings(threshold, warmup=0.2) if threshold else None)
             .add_scheduler_args_if_applicable(pipeline.scheduler, eta=ddim_eta))
-        scale_dim = min(width, height)
-        scale = 512 / scale_dim
 
-        init_width, init_height = trim_to_multiple_of(scale * width, scale * height)
+        # Scale the input width and height for the initial generation
+        # Make their area equivalent to the model's resolution area (e.g. 512*512 = 262144),
+        # while keeping the minimum dimension at least 0.5 * resolution (e.g. 512*0.5 = 256)
+
+        aspect = width / height
+        dimension = 512
+        try:
+            dimension = self.model.unet.config.sample_size * self.model.vae_scale_factor
+        except:
+            pass
+        min_dimension = math.floor(dimension * 0.5)
+
+        model_area = dimension * dimension
+
+        if aspect > 1.0:
+            init_height = max(min_dimension, math.sqrt(model_area / aspect))
+            init_width = init_height * aspect
+        else:
+            init_width = max(min_dimension, math.sqrt(model_area * aspect))
+            init_height = init_width / aspect
+
+        init_width, init_height = trim_to_multiple_of(math.floor(init_width), math.floor(init_height))
+        self.init_width = init_width
+        self.init_height = init_height
```

Comment:
I think it's better in this case to _not_ store these as instance attributes, and let the nested `make_image` function refer to the values from its outer scope instead.

Reason being that I think the instance outlives any one `get_make_image`/`get_image` call, and you don't want the different `get_image` functions to refer to that state that might be changing.

(The whole nested `make_image` function confuses things a lot, and I think it's the next thing on the chopping block after the 2.3 release.)

Reply:
Relevant lines:
https://github.com/invoke-ai/InvokeAI/blob/d57406c4fe6a73f264f8c3663a427ddf3cb07a81/ldm/invoke/generator/base.py#L72-L87

Reply:
Regarding duplication, my testing has shown that any dimension larger than 512 will yield duplication, but those that are too small will cause artifacts in the output. That's how I arrived at 0.5 as a balance and the area calculation as a compromise. Truly if you want to generate an ultra-wide or ultra-tall image and maintain control over the output, the canvas is the only way to go.

Sample outputs, same prompt and seed, fixed maximum dimension of 512:

![512x64](https://user-images.githubusercontent.com/34005131/214611359-9f95189a-22e0-49c0-89be-519c80c08277.png)
![512x128](https://user-images.githubusercontent.com/34005131/214611506-aa832ffd-812b-4689-8314-6e4d40afd18b.png)
![512x192](https://user-images.githubusercontent.com/34005131/214611596-ca571f8a-8149-4e40-9282-8042c6a65f1e.png)
![512x256](https://user-images.githubusercontent.com/34005131/214611744-20ea7776-27ed-4654-bb14-efac2084d0ef.png)
![512x320](https://user-images.githubusercontent.com/34005131/214611912-41a59b4f-5891-49c2-88f0-58883c46dd3d.png)
![512x384](https://user-images.githubusercontent.com/34005131/214612053-e6d08676-ff3d-43d7-b9a8-4e008bdeadaf.png)
![512x448](https://user-images.githubusercontent.com/34005131/214612244-4fcafb44-05bc-4fc5-8420-2550a052d961.png)
![512x512](https://user-images.githubusercontent.com/34005131/214612399-d3c6a7fb-c81f-4b18-bcd6-9c85fdbd4bc7.png)
![448x512](https://user-images.githubusercontent.com/34005131/214612911-9f26a3e3-267b-4188-9ec3-c6d42a3f08d6.png)
![384x512](https://user-images.githubusercontent.com/34005131/214613028-4dc9fc9f-37fa-4cab-a593-6b443a9cbd51.png)
![320x512](https://user-images.githubusercontent.com/34005131/214613139-6f91b5db-7468-44a2-a1b0-c60b10151551.png)
![256x512](https://user-images.githubusercontent.com/34005131/214613248-0dd360a7-ee5a-4ff0-bca2-94689dce249b.png)
![192x512](https://user-images.githubusercontent.com/34005131/214613334-954937f2-608e-4c29-ac6f-ff597acc82e4.png)
![128x512](https://user-images.githubusercontent.com/34005131/214613458-1ef7f998-adad-4c89-b19f-6cae7d3c121a.png)
![64x512](https://user-images.githubusercontent.com/34005131/214613544-c6242511-9b11-407a-9120-bc124558805d.png)